gcloud-java-storage/src/test/java/com/google/cloud/storage/it/ITStorageTest.java

```diff
@@ -877,25 +885,36 @@ public void testBatchRequestFail() {
     Blob remoteBlob = storage.create(blob);
     assertNotNull(remoteBlob);
     BlobInfo updatedBlob = BlobInfo.builder(BUCKET, blobName, -1L).build();
-    BatchRequest batchRequest = BatchRequest.builder()
-        .update(updatedBlob, Storage.BlobTargetOption.generationMatch())
-        .delete(BUCKET, blobName, Storage.BlobSourceOption.generationMatch(-1L))
-        .delete(BlobId.of(BUCKET, blobName, -1L))
-        .get(BUCKET, blobName, Storage.BlobGetOption.generationMatch(-1L))
-        .get(BlobId.of(BUCKET, blobName, -1L))
-        .build();
-    BatchResponse batchResponse = storage.submit(batchRequest);
-    assertEquals(1, batchResponse.updates().size());
-    assertEquals(2, batchResponse.deletes().size());
-    assertEquals(2, batchResponse.gets().size());
-    assertTrue(batchResponse.updates().get(0).failed());
-    assertTrue(batchResponse.gets().get(0).failed());
-    assertFalse(batchResponse.gets().get(1).failed());
-    assertNull(batchResponse.gets().get(1).get());
-    assertTrue(batchResponse.deletes().get(0).failed());
-    assertFalse(batchResponse.deletes().get(1).failed());
-    assertFalse(batchResponse.deletes().get(1).get());
-    assertTrue(remoteBlob.delete());
+    StorageBatch batch = storage.batch();
+    StorageBatchResult<Blob> updateResult =
+        batch.update(updatedBlob, Storage.BlobTargetOption.generationMatch());
+    StorageBatchResult<Boolean> deleteResult1 =
+        batch.delete(BUCKET, blobName, Storage.BlobSourceOption.generationMatch(-1L));
+    StorageBatchResult<Boolean> deleteResult2 = batch.delete(BlobId.of(BUCKET, blobName, -1L));
+    StorageBatchResult<Blob> getResult1 =
+        batch.get(BUCKET, blobName, Storage.BlobGetOption.generationMatch(-1L));
+    StorageBatchResult<Blob> getResult2 = batch.get(BlobId.of(BUCKET, blobName, -1L));
+    batch.submit();
+    try {
+      updateResult.get();
+      fail("Expected StorageException");
+    } catch (StorageException ex) {
+      // expected
+    }
+    try {
+      deleteResult1.get();
+      fail("Expected StorageException");
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [In JUnit4, use the @Test(expected) annotation to denote tests that should throw exceptions](https://www.codacy.com/app/mziccard/gcloud-java/file/2469305504/issues/source?bid=3123103&fileBranchId=3247265#l906)